ComponentKit/Utilities/CKComponentAction.h

```diff
@@ -83,26 +83,25 @@
  {
    _action.send(self, @"hello", 4);
  }
-
  */
 template<typename... T>
-struct CKTypedComponentAction {
+class CKTypedComponentAction : public CKTypedComponentActionBase {
   static_assert(std::is_same<
                 CKTypedComponentActionBoolPack<(std::is_trivially_constructible<T>::value || std::is_pointer<T>::value)...>,
                 CKTypedComponentActionBoolPack<(CKTypedComponentActionDenyType<T>::value)...>
                 >::value, "You must either use a pointer (like an NSObject) or a trivially constructible type. Complex types are not allowed as arguments of component actions.");
-
-  CKTypedComponentAction<T...>() : _internal({}) {};
-  CKTypedComponentAction<T...>(id target, SEL selector) : _internal({CKTypedComponentActionVariantTargetSelector, target, nil, selector})
+public:
+  CKTypedComponentAction<T...>() : CKTypedComponentActionBase() {};
+  CKTypedComponentAction<T...>(id target, SEL selector) : CKTypedComponentActionBase(target, selector)
   {
 #if DEBUG
     std::vector<const char *> typeEncodings;
     CKTypedComponentActionTypeVectorBuild(typeEncodings, CKTypedComponentActionTypelist<T...>{});
     _CKTypedComponentDebugCheckTargetSelector(target, selector, typeEncodings);
 #endif
   }
-
-  CKTypedComponentAction<T...>(const CKComponentScope &scope, SEL selector) : _internal({CKTypedComponentActionVariantComponentScope, nil, scope.scopeHandle(), selector})
+  
+  CKTypedComponentAction<T...>(const CKComponentScope &scope, SEL selector) : CKTypedComponentActionBase(scope, selector)
```

Comment:
@craffert0 I tried to make the change to do the whole
```
using CKTypedComponentActionBase::CKTypedComponentActionBase;
```
when not in debug mode, but it fails to compile, saying that callsites are accessing protected constructors of the base class. Any idea how to fix?